packages/webapp/src/App/Layout/Header/TitleBar/TitleBar.tsx

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { IIconProps } from '@fluentui/react'
+import { PrimaryButton } from '@fluentui/react'
+import type { FC } from 'react'
+import { memo, useCallback, useRef } from 'react'
+import { NavLink } from 'react-router-dom'
+import styled from 'styled-components'
+
+import { Flex } from '~components/Flexbox'
+import { Pages } from '~pages'
+import { useCanRun, useOnGetAllAssetsDownloadInfo } from '~pages/hooks'
+
+const downloadIcon: IIconProps = { iconName: 'Download' }
+
+export const TitleBar: FC = memo(function TitleBar() {
+	const canRun = useCanRun()
+	const downloadAnchorRef = useRef<HTMLAnchorElement>(null)
+	const onGetAllAssetsDownloadInfo = useOnGetAllAssetsDownloadInfo()
+
+	const handleDownload = useCallback(async () => {
+		if (downloadAnchorRef.current) {
+			const info = await onGetAllAssetsDownloadInfo()
+
+			if (info) {
+				downloadAnchorRef.current.href = info.url
+				downloadAnchorRef.current.download = info.alias
+				downloadAnchorRef.current.click()
+			}
+		}
+	}, [downloadAnchorRef, onGetAllAssetsDownloadInfo])
+	return (
+		<Container align="center" justify="space-between">
+			<StyledLink to={Pages.Home.path}>{Pages.Home.name}</StyledLink>
```

Comment:
Maybe we could put the `ProcessingProgress` component somewhere around here?

Reply:
Placed the processing indicator in the title bar.